### PR TITLE
add disableAutoUpgrades as option for controller in provision purpose

### PR DESCRIPTION
**Pull request by Duciwuci**
**How to categorize this issue?**

/area ops-productivity
/kind enhancement

**What would you like to be added**:

Many Ubuntu cloud images have unattended upgrades enabled. To better control the rollout of upgrades we want to disable the unattended upgrade mechanism.

For this we propose to add a new setting to [values.yaml](https://github.com/gardener/gardener-extension-os-ubuntu/blob/master/charts/gardener-extension-os-ubuntu/values.yaml):

```yaml
disableAutoUpgrades: false
```

Setting `disableAutoUpgrades` to `true` disables unattended-upgrades. 
 
According to [ubuntu blog](https://ubuntu.com/blog/ubuntu-updates-best-practices-for-updating-your-instance), setting `APT::Periodic::Unattended-Upgrade “0”;` in `/etc/apt/apt.conf.d/20auto-upgrades` can be used to disable this automatism.

The [`handleProvision`](https://github.com/gardener/gardener-extension-os-ubuntu/blob/master/pkg/controller/operatingsystemconfig/actuator.go#L76) method then should add a new `/etc/apt/apt.conf.d/99auto-upgrades` to the config folder.

**Why is this needed**:

To better control the rollout of security updates or major upgrades, these should be centrally managed. For this, unattended-upgrades inside the images need to be disabled. Instead, we only want to apply upgrades by creating new nodes with an up-to-date image during maintenance windows.

### CODE REVIEW

**Comment by gardener-robot-ci-2** [comment]
Thank you @Duciwuci for your contribution. Before I can start building your PR, a member of the organization must set the required label(s) {'reviewed/ok-to-test'}. Once started, you can check the build status in the PR checks section below.

**Comment by MichaelEischer** [comment]
/unhold

Manual tests using Ubuntu 22.04 were successful. The generated config file successfully disables unattended-upgrades.

**Comment by Duciwuci** [comment]
/unhold manual tests were successful.

@MrBatschner you can retrigger the tests :)

File: pkg/controller/cmd/options.go [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+import "github.com/spf13/pflag"
+
+var (
+	// DisableUnattendedUpgrades is the name of the command line flag to disable unattended upgrades in ubuntu.
+	DisableUnattendedUpgrades = "disable-unattended-upgrades"
+)
+
+// UbuntuOptions are command line options that can be set for ubuntu configuration.
+type UbuntuOptions struct {
+	// DisableUnattendedUpgrades is the flat to disable unattended upgrades in ubuntu.
```
Comment (maboehm) [inline pkg/controller/cmd/options.go:12]:
```suggestion
	// DisableUnattendedUpgrades is the flag to disable unattended upgrades in ubuntu.
```

File: pkg/controller/operatingsystemconfig/add.go [hunk 2]
```diff
@@ -23,13 +23,15 @@ type AddOptions struct {
 	IgnoreOperationAnnotation bool
 	// UseGardenerNodeAgent specifies whether the gardener-node-agent feature is enabled.
 	UseGardenerNodeAgent bool
+	// DisableUnattendedUpgrades is the flat to disable unattended upgrades in ubuntu.
```
Comment (maboehm) [inline pkg/controller/operatingsystemconfig/add.go:26]:
```suggestion
	// DisableUnattendedUpgrades is the flag to disable unattended upgrades in ubuntu.
```

**Comment by gehoern** [comment]
i have been asked to take a look in this ticket. I am maintainer on the Garden Linux side (a debian derivate).

We have unattended-upgrades by default disabled for several reasons:
- default setup only updates non kernel items (most of our hardest issues are kernel related and would need a restart)
- services that are updated are not restarted (they use still old and non updated libraries)
- there is no concept of doing updates in order and reproducible
- Garden Linux is hardened (aka immutable) where those updates make no sense
- booting the latest Gardener Linux image for Gardener fixes all the issues unattended-upgrades would fix if it is used right.

in our opinion running unattended-upgrades makes only sense on longer living nodes. And since there is no upgrade concept (except node rolling with Gardener) it makes no sense to change the behaviour of running nodes.

IMHO plz disable unattended-upgrades as already done in Garden Linux.

**Comment by Duciwuci** [comment]
Thanks @gehoern for your answer!

I have two questions:
* should unattended-upgrades not be configurable? And therefore default `disabled`
* How do you think about our approach, but per default `disabled` or should we tackle another approach?

**Comment by MrBatschner** [comment]
@Duciwuci: Sorry for the delay in reviewing this PR (🤒). In the meatime, to make this a lot cleaner, I merged the PR that removes the old oscommon package. Can you please rebase this PR and remove the references to the old oscommon library? Then, everything should be good to go.

**Comment by Duciwuci** [comment]
> @Duciwuci: Sorry for the delay in reviewing this PR (🤒). In the meatime, to make this a lot cleaner, I merged the PR that removes the old oscommon package. Can you please rebase this PR and remove the references to the old oscommon library? Then, everything should be good to go.

Thanks! I rebased this PR :)

**Review by MrBatschner — APPROVED** [review]
/lgtm

**Pull request merged**